server/routes/polls/add.route.js

```diff
@@ -0,0 +1,57 @@
+const express = require("express");
+const router = express.Router();
+const multer = require("multer");
+const aws = require("aws-sdk");
+const multerS3 = require("multer-s3");
+const Poll = require("../../models/polls");
+
+const s3 = new aws.S3({
+	secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+	accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+});
+
+const upload = multer({
+	storage: multerS3({
+		s3: s3,
+		bucket: "hatchways-social-app",
+		metadata: (req, file, cb) => {
+			cb(null, { fieldName: file.fieldname });
+		},
+		key: function (req, file, cb) {
+			cb(null, Date.now().toString() + "-" + file.originalname);
+		},
+	}),
+});
+
+router.post("/", upload.any(), (req, res) => {
+	const errors = [];
+	if (!req.body.title) errors.push({ title: "Invalid title" });
+	if (!req.body.userId) errors.push({ userId: "Invalid user id" });
+	if (req.files.length < 2)
+		errors.push({ images: "minimum two images required" });
+
+	if (errors.length > 0) return res.status(400).json({ error: errors });
```

Comment:
I believe that @Hakeemmidan was doing validation in a separate file?

Reply:
Yes. There were only two validations and that too, to check if the values exist or not, I was in a conflict whether to do it in a separate file or not.